src/bloomfilter.h

```diff
@@ -42,19 +43,29 @@ int bloomfilter_Update(BloomFilter * bf, char * data, int size);
 BloomFilter * bloomfilter_Copy_Template(BloomFilter * src, char * filename, int perms);
 
 /* A lot of this is inlined.. */
-BTYPE _hash_char(uint32_t hash_seed, Key * key);
+BTYPE _hash_char32(uint32_t hash_seed, Key * key);
+BTYPE _hash_char64(uint32_t hash_seed, Key * key);
 
-BTYPE _hash_long(uint32_t hash_seed, Key * key);
+BTYPE _hash_long32(uint32_t hash_seed, Key * key);
+BTYPE _hash_long64(uint32_t hash_seed, Key * key);
 
 
 static inline int bloomfilter_Add(BloomFilter * bf, Key * key)
 {
-    BTYPE (*hashfunc)(uint32_t, Key *) = _hash_char;
+    BTYPE (*_hash_char)(uint32_t, Key *) = _hash_char32;
+    BTYPE (*_hash_long)(uint32_t, Key *) = _hash_long32;
     register BTYPE mod = bf->array->bits;
     register int i;
     register int result = 1;
     register BTYPE hash_res;
 
+    if (bf->bf_version >= 2 && mod >= HASH64_THRESHOLD) {
+        _hash_char = _hash_char64;
+        _hash_long = _hash_long64;
+    }
+
+    BTYPE (*hashfunc)(uint32_t, Key *) = _hash_char;
```

Comment:
The intent is for hash_X to be inlined (see the comment). Using indirect calls inhibits inlining.

Just have 2 versions of the loop. It's verbose and ugly but will let the compiler inline aggressively.

Reply:
I wasn't sure about this because the library already had (and still has, as right now) a function pointer to change between `hash_char` and `hash_long`, still, duplicating the loop may help the compiler to optimize better since there are only 2 possible values per branch